pkg/plugin/manager.go

```diff
@@ -117,9 +117,6 @@ func (m *Manager) PluginByAlias(alias string) (*Plugin, error) {
 
 // Plugin returns a plugin from the manager.
 func (m *Manager) Plugin(correlationID string) (*Plugin, error) {
-	m.mutex.RLock()
```

Comment:
I think that if you remove the lock, you have a race condition error when:
- I call the Plugin function
- at the same time the plugin is delete

Reply:
nice catch! i will put back the lock